nuxeo-features/nuxeo-platform-audit/nuxeo-platform-audit-core/src/main/java/org/nuxeo/ecm/platform/audit/provider/LatestCreatedUsersOrGroupsPageProvider.java

```diff
@@ -0,0 +1,97 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     "Guillaume Renard"
+ */
+
+package org.nuxeo.ecm.platform.audit.provider;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.directory.Session;
+import org.nuxeo.ecm.directory.api.DirectoryService;
+import org.nuxeo.ecm.platform.audit.api.LogEntry;
+import org.nuxeo.ecm.platform.query.api.AbstractPageProvider;
+import org.nuxeo.ecm.platform.query.api.PageProvider;
+import org.nuxeo.ecm.platform.query.api.PageProviderService;
+import org.nuxeo.ecm.platform.usermanager.UserManager;
+import org.nuxeo.ecm.platform.usermanager.UserManagerImpl;
+import org.nuxeo.runtime.api.Framework;
+
+/**
+ * @since 9.3
+ */
+public class LatestCreatedUsersOrGroupsPageProvider extends AbstractPageProvider<DocumentModel> {
+
+    private static final long serialVersionUID = 1L;
+
+    public final static String LATEST_CREATED_USERS_OR_GROUPS_PROVIDER = "LATEST_CREATED_USERS_OR_GROUPS_PROVIDER";
+
+    public final static String LATEST_AUDITED_CREATED_USERS_OR_GROUPS_PROVIDER = "LATEST_AUDITED_CREATED_USERS_OR_GROUPS_PROVIDER";
+
+    protected List<DocumentModel> currentPage;
+
+    @Override
+    public List<DocumentModel> getCurrentPage() {
+        if (currentPage == null) {
+            currentPage = new ArrayList<DocumentModel>();
+            PageProviderService pps = Framework.getService(PageProviderService.class);
+            PageProvider<?> pp = pps.getPageProvider(LATEST_AUDITED_CREATED_USERS_OR_GROUPS_PROVIDER, null,
+                    getPageSize(), getCurrentPageIndex(), getProperties(), new Object[] { null });
+            @SuppressWarnings("unchecked")
+            List<LogEntry> entries = (List<LogEntry>) pp.getCurrentPage();
+            if (entries != null) {
+                UserManager um = Framework.getService(UserManager.class);
+                DirectoryService directoryService = Framework.getService(DirectoryService.class);
+                try (Session userDir = directoryService.open(um.getUserDirectoryName(), null)) {
+                    for (LogEntry e : entries) {
+                        String id = (String) e.getExtendedInfos().get("id").getSerializableValue();
+                        if (StringUtils.isNotBlank(id)) {
+                            DocumentModel doc;
+                            if (UserManagerImpl.GROUPCREATED_EVENT_ID.equals(e.getEventId())) {
+                                doc = um.getGroupModel(id);
+                            } else if (UserManagerImpl.USERCREATED_EVENT_ID.equals(e.getEventId())) {
+                                doc = um.getUserModel(id);
+                                if (doc == null) {
+                                    continue;
+                                }
```

Comment:
This `if (doc == null)` is redundant with code a few lines below.

Reply:
ok